apps/builder/app/builder/features/style-panel/shared/css-value-input/css-value-input.tsx

```diff
@@ -318,8 +318,15 @@ export const CssValueInput = ({
       return;
     }
 
+    const parsedValue = parseIntermediateOrInvalidValue(property, value);
+
+    if (parsedValue.type === "invalid") {
+      props.onChange(parsedValue);
+      return;
+    }
```

Comment:
This is a fantastic piece of logic, never seen anything better than this

Reply:
Every time I see it, my eyes water with delight.